client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/RuleScope.java

```diff
@@ -59,6 +59,15 @@
         };
     }
 
+    private static final DeprecationHandler DEPRECATION_HANDLER = new DeprecationHandler() {
```

Comment:
is this the only use of DeprecationHandler in the code? There is an instance of this handler in the RestHighLevelClient as well, so it might make more sense to use the one there?

Reply:
It is actually the only use. Though we could reuse the one in RestHighLevelClient, it is an odd dependency direction. I am tempted to duplicate for now. If we hit another we can refactor.